src/index.js

```diff
@@ -1,14 +1,18 @@
 import behaviorSubject from 'callbag-behavior-subject'
-import { useRef } from 'react'
+import { useRef, useEffect } from 'react'
 
 export default function useValueCallbag(value) {
-  const value$Ref = useRef(null)
+  const value$Ref = useRef()
 
-  if (value$Ref.current === null) {
+  if (!value$Ref.current) {
     value$Ref.current = behaviorSubject(value)
-  } else {
-    value$Ref.current(1, value)
   }
 
+  useEffect(() => {
+    value$Ref.current(1, value)
+  }, [value])
+
+  useEffect(() => () => value$Ref.current(2), [])
```

Comment:
Since the callbag is internal only I think it's safe to just rely on GC (at least for most cases). The problem is that React doesn't really allow us to hook into "unmounting". For instance, this would be called during Fast Refresh, which doesn't unmount components and this, in turn, would violate the callbag spec as we'd continue delivering `1` notifications after already giving `2` to the callbag.

Reply:
> On this front I just followed React's docs for having a cleanup on unmount.

Ye, the docs are not precise there 😢 

> I'm also having difficulty understanding the example you mentioned, does that mean in some situations the ref is kept in tact but the cleanup callback is invoked?

Exactly.

> Could you provide some further clues on how to reproduce this buggy behavior?

I don't have time right now to prepare a proper demo but it should go like this:
1. initialize a fresh CRA project
- install required deps 
- copy the source code of this project into it so you can play around with it easier
2. create a component utilizing the `useValueCallbag`
3. trigger some updates
4. edit a component file (add a `console.log` or smth) **without** refreshing your browser

Reply:
perfect! I will try to reproduce it and see if we can maintain a logical terminate signal without much complexity. If that leads to too much complexity, then I'll remove the terminate signal handler.

Reply:
ok this was rather easy to reproduce. a (relatively) simple work-around is to also reset the value of the ref whenever the callbag is terminated, and create and return a new subject when the hook is re-executed.

one down-side-_ish_ with this approach is that effects subscribing to a stream **MUST** include it as a dependency. for example:

```js
const stream = useValueCallbag(...)
useEffect(() => pipe(stream, ...), [])
```

in this example, when a new subject is created due to fast-refresh, it is NOT picked up by the effect (and so not subscribed to). however, either of these versions works properly:

```js
useEffect(() => pipe(stream, ...))
```
```js
useEffect(() => pipe(stream, ...), [stream])
```

since it also appears that standard linters would give a warning in the first case but accept any of the latter two cases, I wouldn't worry about this scenario.

p.s. also I have no idea how to write a test for this, honestly not sure if it is possible ([_yet_](https://github.com/testing-library/react-testing-library/issues/849)).

Reply:
> p.s. also I have no idea how to write a test for this, honestly not sure if it is possible (yet).

Ye, don't bother with writing any tests related to this.